mesonbuild/mesonlib.py

```diff
@@ -1520,6 +1520,15 @@ def discard(self, value: _T) -> None:
         if value in self.__container:
             del self.__container[value]
 
+    def move_to_end(self, value: _T, last: bool = True) -> None:
+        # Mypy does not know about move_to_end, because it is not part of MutableMapping
+        self.__container.move_to_end(value, last) # type: ignore
+
+    def popitem(self, last: bool = True) -> _T:
+        # Mypy does not know about the last argument, because it is not part of MutableMapping
+        item, _ = self.__container.popitem(last)  # type: ignore
+        return item
+
```

Comment:
You're right that I should have named it `pop`, if that's what you mean.

However, if you mean that the extra `last` argument is a problem, OrderedSet would remain a superset of the set thanks to default parameters. OrderedDict does the same with dict by adding `popitem(last=true)`.

Reply:
In particular, something like this:

```diff
--- a/mesonbuild/mesonlib.py
+++ b/mesonbuild/mesonlib.py
@@ -1540,10 +1540,20 @@ class OrderedSet(T.MutableSet[_T]):
     def add(self, value: _T) -> None:
         self.__container[value] = None
 
+    def addfirst(self, value: _T) -> None:
+        self.add(value)
+        # Mypy does not know about move_to_end, because it is not part of MutableMapping
+        self.__container.move_to_end(value, First) # type: ignore
+
     def discard(self, value: _T) -> None:
         if value in self.__container:
             del self.__container[value]
 
+    def popfirst(self) -> _T:
+        # Mypy does not know about the last argument, because it is not part of MutableMapping
+        item, _ = self.__container.popitem(False)  # type: ignore
+        return item
+
     def update(self, iterable: T.Iterable[_T]) -> None:
         for item in iterable:
             self.__container[item] = None
```

wouldn't remove the need for Mypy annotation. In both cases `OrderedSet` would be a superset of `set`; however, the version in the pull request also makes `OrderedSet` look more like `OrderedDictionary`.

Reply:
@dcbaker Ping, I don't understand the objection so I cannot fix the pull request. I can always use myself an OrderedDict with `None` values, but I'd rather avoid that.